### ISSUE TITLE
Windows Hello dialog position at top-left of the screen if main window is hidden

### ISSUE THREAD

**Issue by mateuszdrab** [sirAndros/KeePassWinHello#54]
First of all, thanks for making this awesome plugin that essentially made me finally switch to Keepass. It seems to work better than other key provider plugins as this doesn't seem to work as a key provider, as such I can still have programmatic access to the DB with the standard set of keys.

Onto the point...

**Describe the bug**
This seems to be a small bug with the handling of the Windows Hello dialog. When Keepass is minimised or in tray and an unlock is attempted (either via the context menu or through another plugin requesting an unlock), the form appears at what top left corner of the screen.

This makes sense as there is no parent window to center upon, it would be much better if the form could be displayed in the middle of the screen or perhaps at the last position (the position could be saved in memory).

Would this be easy to solve? I can try taking a look at the code myself though it will probably take me some time to setup the environment to compile the plugin.

**Expected behavior**
The window appears at the center of the screen when main keepass window is not visible.

**Desktop (please complete the following information):**
 - Windows 10 x86 1909
 - KeePass Version 2.44
 - Plugin Version 3.1.1.0
 - Hardware info Screen resolution = 4k
 
 **KeePass settings and enviroment**
 - Is secure desktop enabled = no
 - Is KeePass running under Administrator (in elevated process) = no
 - Installed plugins (if any) KeePassOtp, KeeAgent (this triggers winhello when key is needed)

Thanks!

**Comment by musm** [comment]
I also get this and it's pretty annoying

**Comment by SilveIT** [comment]
I agree. Getting the "Windows Security" dialog in the background is very annoying. I guess there's a problem with NCRYPT_WINDOW_HANDLE_PROPERTY?

**Comment by StefanKrapf** [comment]
Me too. I have 3 monitors and the window is appearing at the left screen top left.
But the window has no focus too! If I have many windows opened, often the Windows Hello window appears behind this windows.
It would be nice this could be fixed.

**Comment by DIDoS** [comment]
I guess an option to always show it as "top" would be appropriate since sometimes the call might not be from kp itself but from a browser or other application, with an unknown HWND. If activated, it would need to get the HWND of the TopWindow (GetTopWindow) and use it when setting the NCRYPT_WINDOW_HANDLE_PROPERTY as mentioned by @SilveIT

**Comment by fritut08** [comment]
> 
> 
> Me too. I have 3 monitors and the window is appearing at the left screen top left.
> But the window has no focus too! If I have many windows opened, often the Windows Hello window appears behind this windows.
> It would be nice this could be fixed.

I agree with @StefanKrapf. When I activate auto-type via Ctrl+Alt+A the WinHello-Window opens in background. It should be in foreground.

**Comment by davidegiacometti** [comment]
Hi,
not sure if the plugin still supported but I have created a PR for fix this.

**Comment by ArkitectNaut** [comment]
> When I activate auto-type via Ctrl+Alt+A the WinHello-Window opens in background

Same exact problem. The most 'annoying' part of it is that for the finger-print scanner to work, I must first selected the window, adding friction to the whole interaction.

**Comment by sirAndros** [comment]
Hi! Please check if the issue exists in v3.3

**Comment by shell-surfer** [comment]
Hi, Windows-Hello still does not have focus in the following cases.

1. 'Minimize to tray instead of taskbar' option is turned on under 'Interface 1' > 'Main window' in keypass settings. Locking the database when keypass window is visible and minimising sends keypass to tray. (If it is minimised to the the taskbar, windows hello does open in focus). Activating auto-type via Ctrl+Alt+A the WinHello-Window opens in background.

2. Also happens when 'Start minimized and locked' option is turned on under 'Advanced' > 'Start and exit', while 'Minimize to tray instead of taskbar' option is turned on and keypass starts to tray. Activating auto-type via Ctrl+Alt+A the WinHello-Window opens in background.

Also noticed that if you lock it from the tray while the keypass window is not visible, windows hello does open in focus when activating auto-type.
